image/docker_schema2.go

```diff
@@ -177,12 +177,23 @@ func (m *manifestSchema2) UpdatedImage(options types.ManifestUpdateOptions) (typ
 }
 
 func (m *manifestSchema2) convertToManifestOCI1() (types.Image, error) {
-	// Create a copy of the descriptor.
-	config := m.ConfigDescriptor
+	configBlob, err := m.ConfigBlob()
+	if err != nil {
+		return nil, err
+	}
+	configOCI := &imgspecv1.Image{}
```

Comment:
It’s non-obvious what this does. This would benefit from a comment, saying something like “The two versions of config are mostly compatible, but v2s2 has an extra $name field; this unmarshal+re-marshal effectively drops it.”